lib/client/connection.js

```diff
@@ -714,3 +692,43 @@ Connection.prototype._handleSnapshotFetch = function(error, message) {
   delete this._snapshotRequests[message.id];
   snapshotRequest._handleResponse(error, message);
 };
+
+Connection.prototype._handleLegacyInit = function(message) {
```

Comment:
Let's move `_initializeHandshake` out of here to save half a round trip, and make it easier to remove legacy behaviour in future.